x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/components/expandable_section/expandable_section_results.tsx

```diff
@@ -127,6 +186,104 @@ export const ExpandableSectionResults: FC<ExpandableSectionResultsProps> = ({
   );
   const analysisType =
     jobConfig && jobConfig.analysis ? getAnalysisType(jobConfig.analysis) : undefined;
+
+  const generateDiscoverUrl = useCallback(
+    async (rowIndex: number) => {
+      const item = tableItems[rowIndex];
+
+      if (discoverLocator !== undefined) {
+        const url = await discoverLocator.getRedirectUrl({
+          indexPatternId: dataViewId,
+          timeRange: data.query.timefilter.timefilter.getTime(),
+          filters: data.query.filterManager.getFilters(),
+          query: {
+            language: SEARCH_QUERY_LANGUAGE.KUERY,
+            // Filter for all visible column values of supported types - except the results field values
+            query: indexData.visibleColumns
+              .filter(
+                (column) =>
+                  item[column] !== undefined &&
+                  (typeof item[column] === 'string' || typeof item[column] === 'number') &&
+                  !column.includes(resultsField!)
+              )
+              .map((column) => `${escapeKuery(column)}:${escapeKuery(String(item[column]))}`)
+              .join(' and '),
+          },
+        });
+
+        return url;
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [indexData?.visibleColumns, discoverLocator, dataViewId, resultsField, tableItems]
```

Comment:
can `data` be included here to remove the need to suppress this error?

Reply:
Included in 1664d294c98a5c1b8c9e86cfc426be49b467bd40